src/runtime_src/hip/core/graph.cpp

```diff
@@ -134,7 +134,7 @@ init(std::shared_ptr<graph> graph)
 graph_exec::
 graph_exec(std::shared_ptr<graph> graph)
   : command(type::graph_exec)
-  , m_node_exec_list(init(graph))
+  , m_node_exec_list(init(std::move(graph)))
```

Comment:
I don't see the shared ownership of graph.   The graph is not stored anywhere here, so the signature of this function should be either `const std::shared_ptr<graph>&` or `graph* graph`, ditto for the init function.